spec/controllers/sign_up/select_email_controller_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe SignUp::SelectEmailController do
+  describe '#create' do
+    let(:email) { 'michael.motorist@email.com' }
+    let(:email2) { 'michael.motorist2@email.com' }
+    let(:email3) { 'david.motorist@email.com' }
+    let(:user) { create(:user) }
+
+    before do
+      user.email_addresses = []
+      user.email_addresses.create(email: email, confirmed_at: Time.zone.now)
+      user.email_addresses.create(email: email2, confirmed_at: Time.zone.now)
```

Comment:
Minor:

- I don't think the assignment does anything?
- We can use the [email address spec factory](https://github.com/18F/identity-idp/blob/main/spec/factories/email_addresses.rb) to avoid dealing with details like `confirmed_at`

```suggestion
      create(:email_address, user:, email: email)
      create(:email_address, user:, email: email2)
```

Reply:
Do you mean the empty array? I was thinking I would make sure the test user only had the email addresses that follow to make the test more sterile.

Reply:
That wouldn't actually destroy any email addresses associated with the user, since that's only referencing a local copy. You'd need to `user.email_addresses.destroy_all` to actually destroy the email addresses.

But since the user is created in the scope of the test, we should be able to have full control over whether the user has any email addresses when we create it (example).